src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -303,9 +295,8 @@ private void putInternal(Model model, ActionListener<IndexResponse> listener, Do
                 parameters.put(KNNConstants.MODEL_BLOB_PARAMETER, base64Model);
             }
 
-            final IndexRequestBuilder indexRequestBuilder = ModelDao.runWithStashedThreadContext(
-                () -> client.prepareIndex(MODEL_INDEX_NAME)
-            );
+            IndexRequestBuilder indexRequestBuilder = client.prepareIndex(MODEL_INDEX_NAME);
```

Comment:
same as above.

Reply:
we do context stashing at the transport action level, keeping this functionality here is redundant

Reply:
Thanks this make sense